src/main/java/com/mongodb/jdbc/MongoConnection.java

```diff
@@ -82,19 +82,35 @@ public MongoConnection(ConnectionString cs, String database, Level logLevel, Fil
                                 .append(".")
                                 .append(MongoDriver.MINOR_VERSION)
                                 .toString();
+        StringBuilder appName = new StringBuilder(MongoDriver.NAME);
 
         // Log the driver name and version
         logger.log(
                 Level.INFO, "Connecting using " + MongoDriver.MONGOSQL_DRIVER_NAME + " " + version);
 
-        this.mongoClientSettings = MongoClientSettings.builder().applyConnectionString(cs).build();
-        mongoClient =
-                MongoClients.create(
-                        mongoClientSettings,
-                        MongoDriverInformation.builder()
-                                .driverName(MongoDriver.NAME)
-                                .driverVersion(version)
-                                .build());
+        MongoDriverInformation.Builder mdiBuilder;
+        if (connectionProperties.getClientInfo() != null
+                && connectionProperties.getClientInfo().length == 2) {
+            appName.append('|').append(connectionProperties.getClientInfo()[0]);
+            MongoDriverInformation driverInfoWithClientInfo =
+                    MongoDriverInformation.builder()
+                            .driverName(connectionProperties.getClientInfo()[0])
+                            .driverVersion(connectionProperties.getClientInfo()[1])
+                            .build();
+            mdiBuilder = MongoDriverInformation.builder(driverInfoWithClientInfo);
+        } else {
+            mdiBuilder = MongoDriverInformation.builder();
+        }
+        MongoDriverInformation mongoDriverInformation =
+                mdiBuilder.driverName(MongoDriver.NAME).driverVersion(version).build();
+
+        this.mongoClientSettings =
+                MongoClientSettings.builder()
+                        .applicationName(appName.toString())
```

Comment:
The driver version and connector version will not be in `appName`. Is this intentional?

Reply:
Also, just saw this while connecting with mongosh : appName=mongosh+1.2.2
What about we follow the convention and use something similar like `driverName+driverVersion|tableau+connectorVersion`?

Reply:
Sounds good, I'll add the version to the appName.

Reply:
Here's an example output: `"appName": "mongodb-jdbc+2.0.0-beta-2-7-g1c59d6b-dirty|tableau-connector+f8807d3"`